internal/controller/config.go

```diff
@@ -0,0 +1,123 @@
+package controller // import "go.opentelemetry.io/ebpf-profiler/internal/controller"
```

Comment:
What about naming this internal package `args` instead of `controller`?
From a controller I expect something different, than providing a general configuration and validate it.
As this package is likely reused in the OTel profiling controller part, it should not get mixed and suggest something it doesn't do. 
```suggestion
package args // import "go.opentelemetry.io/ebpf-profiler/internal/args"
```

Reply:
This is meant to be the first of a few PRs, with the intent to extract the setup of `main.go` into a shared package.
I named this package `controller`, because what it will do in the end is run the agent, either standalone, or as a collector receiver.